worker-queue-rabbit/src/main/java/com/github/workerframework/queues/rabbit/WorkerQueueConsumerImpl.java

```diff
@@ -100,20 +118,72 @@ public void processDelivery(Delivery delivery)
             taskInformation.incrementResponseCount(true);
-            publisherEventQueue.add(new WorkerPublishQueueEvent(delivery.getMessageData(), retryRoutingKey, taskInformation,
-                    Collections.singletonMap(RabbitHeaders.RABBIT_HEADER_CAF_WORKER_REJECTED, REJECTED_REASON_TASKMESSAGE)));
+            final var publishHeaders = new HashMap<String, Object>();
+            publishHeaders.put(RabbitHeaders.RABBIT_HEADER_CAF_WORKER_REJECTED, REJECTED_REASON_TASKMESSAGE);
+            taskMessageStorageRefOpt.ifPresent(s -> publishHeaders.put(RABBIT_HEADER_CAF_MINIMIZATION_ID, s));
+            publisherEventQueue.add(new WorkerPublishQueueEvent(inboundByteArray, retryRoutingKey, taskInformation, publishHeaders));
         } catch (TaskRejectedException e) {
-            LOG.warn("Message {} rejected as a task at this time, returning to queue", taskInformation.getInboundMessageId(), e);
+            final RabbitTaskInformation taskInformation = new RabbitTaskInformation(
+                String.valueOf(inboundMessageId), 
+                isPoison
+            );
+            LOG.warn("Message {} rejected as a task at this time, returning to queue", inboundMessageId, e);
             taskInformation.incrementResponseCount(true);
-            publisherEventQueue.add(new WorkerPublishQueueEvent(delivery.getMessageData(), delivery.getEnvelope().getRoutingKey(),
-                    taskInformation));
+            publisherEventQueue.add(new WorkerPublishQueueEvent(inboundByteArray, routingKey, taskInformation, delivery.getHeaders()));
+        }
+    }
+
+    /**
+     * Deserialize the task message from the delivery message data. If the task message is minimized, retrieve it from the data store.
+     * 
+     * @param inboundMessageId
+     * @param deliveryMessageData
+     * @param taskMessageStorageRefOpt
+     * @return
+     * @throws InvalidTaskException
+     */
+    private TaskMessage deserializeTaskMessage(
+        final long inboundMessageId,
+        final byte[] deliveryMessageData, 
+        final Optional<String> taskMessageStorageRefOpt
+    ) throws InvalidTaskException {
+        try {
+            if (taskMessageStorageRefOpt.isPresent()) {
+                final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                try (final var inputStream = dataStore.retrieve(taskMessageStorageRefOpt.get())) {
+                    final byte[] buffer = new byte[1024];
+                    int length;
+                    while ((length = inputStream.read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, length);
+                    }
+                }
+                return codec.deserialise(outputStream.toByteArray(), TaskMessage.class, DecodeMethod.LENIENT);
+            }
+           return codec.deserialise(deliveryMessageData, TaskMessage.class, DecodeMethod.LENIENT);
+        } catch (final IOException | CodecException | DataStoreException e) {
```

Comment:
If you apply the above suggestion then IOException can be removed.

```suggestion
        } catch (final CodecException | DataStoreException e) {
```

Reply:
done